protocol/ballot.go

```diff
@@ -10,7 +10,7 @@ import "fmt"
 // (counter, ID)."
 type Ballot struct {
 	Counter uint64
-	ID      uint64
+	ID      string
```

Comment:
What's the motivation to change to string for the id? The latest version (v3) still mentions numerical IDs, has this been changed since?

Reply:
The only requirement for the ID component of the Ballot is that it's capable of being compared to other IDs to break counter ties. I think strings satisfy that just as well as uint64s, and are easier to assign to ephemeral proposer nodes, as well as easier to read in logs. So, just a usability improvement!